app/controllers/users/sessions_controller.rb

```diff
@@ -35,4 +43,38 @@ def confirm_auth
       render json: ["Incorrect password. Please enter your #{Settings.general.name} password."], status: :unprocessable_entity
     end
   end
+
+  def send_otp
+    respond_to do |format|
+      format.json do
+        email = params[:email]
+        user = User.find_for_authentication(email: email)
+
+        return render status: 404, json: ['Email not recognized.'] unless user
+        return render status: 422, json: ['OTP not required'] unless user.otp_required_for_login?
+
+        UserMailer.otp_requested(user).deliver_now
```

Comment:
This should only send if the user and password are valid. Otherwise, it allows a random person to spam a registered user with unusable otp emails.